x-pack/plugins/ml/server/models/data_recognizer/data_recognizer.ts

```diff
@@ -933,28 +944,115 @@ export class DataRecognizer {
     }
   }
 
-  // ensure the model memory limit for each job is not greater than
-  // the max model memory setting for the cluster
-  async updateModelMemoryLimits(moduleConfig: Module) {
+  /**
+   * Provides a time range of the last 3 months of data
+   */
+  async getFallbackTimeRange(
+    timeField: string,
+    query?: any
+  ): Promise<{ start: number; end: number }> {
+    const fieldsService = fieldsServiceProvider(this.callAsCurrentUser);
+
+    const timeFieldRange = await fieldsService.getTimeFieldRange(
+      this.indexPatternName,
+      timeField,
+      query
+    );
+
+    return {
+      start: timeFieldRange.end.epoch - moment.duration(3, 'months').asMilliseconds(),
+      end: timeFieldRange.end.epoch,
+    };
+  }
+
+  /**
+   * Ensure the model memory limit for each job is not greater than
+   * the max model memory setting for the cluster
+   */
+  async updateModelMemoryLimits(
+    moduleConfig: Module,
+    estimateMML: boolean = false,
+    start?: number,
+    end?: number
+  ) {
     const { limits } = await this.callAsCurrentUser('ml.info');
+
     const maxMml = limits.max_model_memory_limit;
-    if (maxMml !== undefined) {
-      // @ts-ignore
-      const maxBytes: number = numeral(maxMml.toUpperCase()).value();
-
-      if (Array.isArray(moduleConfig.jobs)) {
-        moduleConfig.jobs.forEach(job => {
-          const mml = job.config?.analysis_limits?.model_memory_limit;
-          if (mml !== undefined) {
-            // @ts-ignore
-            const mmlBytes: number = numeral(mml.toUpperCase()).value();
-            if (mmlBytes > maxBytes) {
-              // if the job's mml is over the max,
-              // so set the jobs mml to be the max
-              job.config.analysis_limits!.model_memory_limit = maxMml;
-            }
+    // @ts-ignore
+    const maxBytes: number = numeral(maxMml.toUpperCase()).value();
```

Comment:
I don't think it's safe to assume `max_model_memory_limit` is set. I don't have it set, which might explain this error I see when trying to install the apache_ecs module:

![image](https://user-images.githubusercontent.com/7405507/77320675-ea856c80-6d08-11ea-84b6-64b67f35039b.png)

Reply:
it's probably worth changing 
`const { limits } = await this.callAsCurrentUser('ml.info');`
to
`const { limits } = await this.callAsCurrentUser<MlInfoResponse>('ml.info');`
to get the correct response for `ml.info`

Reply:
Fixed in https://github.com/elastic/kibana/pull/60656/commits/80eaeef10586db07fa3e4580498badbb8d2f0df7